nnpdf_data/nnpdf_data/commondata/ATLAS_PH_8TEV/filter.py

```diff
@@ -0,0 +1,259 @@
+            errors[f"sys_corr_{j + 1}"] = float(unc)
+
+        combined_errors.append(errors)
+
+    return combined_errors
+
+
+def dump_commondata(kinematics: list, data: list, errors: dict, obs: str) -> None:
+    """Function that generates and writes the commondata files.
+
+    Parameters
+    ----------
+    kinematics: list
+        list containing the kinematic values
+    data: list
+        list containing the central values
+    errors: dict
+        Dictionary containing the different errors
+    obs: str
+        Name to append to the file names
+    """
+
+    if "statistics" in errors:
+        error_definition = {
+            "stat": {
+                "description": "Uncorrelated statistical uncertainties",
+                "treatment": errors["statistics"].loc["treatment"].iloc[0],
+                "type": errors["statistics"].loc["type"].iloc[0],
+            }
+        }
+    else:
+        error_definition = {}
+
+    n_sys = errors["systematics"].shape[1]
+    for i in range(n_sys):
+
+        error_definition[f"sys_corr_{i + 1}"] = {
+            "description": errors["systematics"].columns[i],
+            "treatment": errors["systematics"].loc["treatment"].iloc[i],
+            "type": errors["systematics"].loc["type"].iloc[i],
+        }
+
+    errors_formatted = format_uncertainties(errors)
+    with open(f"data_{obs}.yaml", "w") as file:
+        yaml.dump({"data_central": data.tolist()}, file, sort_keys=False)
+
+    with open(f"kinematics_{obs}.yaml", "w") as file:
+        yaml.dump({"bins": kinematics}, file, sort_keys=False)
+
+    with open(f"uncertainties_{obs}.yaml", "w") as file:
+        yaml.dump(
+            {"definitions": error_definition, "bins": errors_formatted}, file, sort_keys=False
+        )
+
+
+def main_filter() -> None:
+    """
+    Main function that reads the HepData yaml files and generates the commondata files
+    """
+
+    yaml_content_data = [load_yaml(table_id=i, version=1) for i in range(1, 5)]
+    uncertainties_all = pd.DataFrame()
+    central_values_all = np.array([])
+    kinematics_all = []
+    n_datapoints = [22, 21, 18, 18]
+    mid_rapidities = [0.3, 0.985, 1.685, 2.09]
+    for i, yaml_content in enumerate(yaml_content_data):
+        kinematics = get_kinematics(
+            yaml_content, bin_index=range(n_datapoints[i]), mid_rap=mid_rapidities[i]
+        )
+        central_values, uncertainties = get_errors(yaml_content, bin_index=range(n_datapoints[i]))
+        uncertainties_all = pd.concat([uncertainties_all, uncertainties])
+        central_values_all = np.concatenate([central_values_all, central_values])
+        kinematics_all += kinematics
+
+    uncertainties_all.index = [f"bin {i}" for i in range(uncertainties_all.shape[0])]
+
+    n_sources = uncertainties_all.shape[1]
+    sys_types = {
+        "treatment": ["ADD"] + ["MULT"] * (n_sources - 1),
+        "type": ["UNCORR"] * (n_sources - 1) + ["ATLASLUMI15"],
```

Comment:
```suggestion
        "type": ["UNCORR"] * (n_sources - 1) + ["ATLASLUMI12"],
```

This is part of the 2012 data https://github.com/NNPDF/nnpdf/pull/2205#issuecomment-2605365260